canary.go

```diff
@@ -20,6 +21,9 @@ type TunnelTest struct {
 }
 
 func main() {
+	_, cancel := context.WithDeadline(context.Background(), time.Now().Add(60*time.Second))
```

Comment:
This doesn't do anything alone.. we need to pass the context created to the various functions. (Dial, Command etc..):
  https://golang.org/pkg/net/#Dialer.DialContext
  https://golang.org/pkg/os/exec/#CommandContext

Reply:
Thanks that makes more sense now